editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/pages/Wsdl2RestWizardSecondPage.java

```diff
@@ -51,61 +48,75 @@ public void createControl(Composite parent) {
+		javaPathBinding = createBinding(javaPathTextControl, "destinationJava", new PathValidator());
+		ControlDecorationSupport.create(javaPathBinding, SWT.LEFT | SWT.TOP);
+
+		camelPathBinding = createBinding(camelPathTextControl, "destinationCamel", new PathValidator());
+		ControlDecorationSupport.create(camelPathBinding, SWT.LEFT | SWT.TOP);
+
+		Binding targetAddressBinding = createBinding(targetAddressText, "targetServiceAddress", new TargetURLValidator()); //$NON-NLS-1$
+		ControlDecorationSupport.create(targetAddressBinding, SWT.LEFT | SWT.TOP);
+
+		Binding targetRestAddressBinding = createBinding(targetRestAddressText, "targetRestServiceAddress", new TargetURLValidator()); //$NON-NLS-1$
+		ControlDecorationSupport.create(targetRestAddressBinding, SWT.LEFT | SWT.TOP);
+	}
+
+	private Text createDestinationCamelFolderControl(Composite composite) {
+		Text camelPathTextControl = createLabelAndText(composite, UIMessages.wsdl2RestWizardSecondPageCamelFolderLabel, 2);
+		Button outPathBrowseButton = createButton(composite, "..."); //$NON-NLS-1$
+		outPathBrowseButton.addSelectionListener(new SelectionAdapter() {
 			@Override
 			public void widgetSelected(SelectionEvent e) {
 				// browse
 				IProject project = ResourcesPlugin.getWorkspace().getRoot().getProject(getOptionsFromWizard().getProjectName());
 				String path = selectFolder(project);
 				if (path != null) {
-					getOptionsFromWizard().setDestinationJava(path);
-					javaPathTextControl.notifyListeners(SWT.Modify, new Event());
+					getOptionsFromWizard().setDestinationCamel(path);
+					camelPathTextControl.notifyListeners(SWT.Modify, new Event());
+					camelPathBinding.updateModelToTarget();
 				}
 			}
 		});
+		return camelPathTextControl;
+	}
+
+	private Text createDestinationJavaFolderControl(Composite composite) {
+		Text javaPathTextControl = createLabelAndText(composite, UIMessages.wsdl2RestWizardSecondPageJavaFolderLabel, 2);
+		Button javaPathBrowseBtn = createButton(composite, "..."); //$NON-NLS-1$
+		javaPathBrowseBtn.addSelectionListener(new SelectionAdapter() {
 
-		Text camelPathTextControl = createLabelAndText(composite, UIMessages.wsdl2RestWizardSecondPageCamelFolderLabel, 2);
-		Button outPathBrowseButton = createButton(composite, "..."); //$NON-NLS-1$
-		outPathBrowseButton.addSelectionListener(new SelectionAdapter() {
 			@Override
 			public void widgetSelected(SelectionEvent e) {
 				// browse
 				IProject project = ResourcesPlugin.getWorkspace().getRoot().getProject(getOptionsFromWizard().getProjectName());
 				String path = selectFolder(project);
 				if (path != null) {
-					getOptionsFromWizard().setDestinationCamel(path);
-					camelPathTextControl.notifyListeners(SWT.Modify, new Event());
+					getOptionsFromWizard().setDestinationJava(path);
+					javaPathTextControl.notifyListeners(SWT.Modify, new Event());
+					javaPathBinding.updateModelToTarget();
```

Comment:
here is the second and last real behavior modification (other parts are just refactoring)